frame/support/procedural/src/lib.rs

```diff
@@ -778,6 +781,243 @@ pub fn storage_alias(_: TokenStream, input: TokenStream) -> TokenStream {
 		.into()
 }
 
+/// This attribute can be used to derive a full implementation of a trait based on a local
+/// partial impl and an external impl containing defaults that can be overriden in the local
+/// impl.
+///
+/// ## Usage
+///
+/// The attribute should be attached to an impl block (strictly speaking a `syn::ItemImpl`) for
+/// which we want to inject defaults in the event of missing trait items in the block.
+///
+/// The attribute takes two arguments separated by an `as` with the general form:
+///
+/// ```ignore
+/// #[derive_impl(foreign_path as disambiguation_path)]
+/// impl SomeTrait for SomeStruct {
+/// 	...
+/// }
+/// ```
+///
+/// The `foreign_path` should be the path to an impl registered via
+/// [`#[register_default_impl]`](`macro@register_default_impl`) that contains the default trait
+/// items we want to potentially inject.
+///
+/// The `disambiguation_path` should be the path to a trait that will be used to qualify all
+/// default entries that are injected into the local impl. For example if your `foreign_path`
+/// is `some::path::TestTrait` and your `disambiguation_path` is `another::path::DefaultTrait`,
+/// any items injected into the local impl will be qualified as `<some::path::TestTrait as
+/// another::path::DefaultTrait>::specific_trait_item`.
+///
+/// Optionally, you may omit the `as disambiguation_path` portion, in which case the
+/// `disambiguation_path` will internally default to `A` from the `impl A for B` part of the
+/// foreign impl. This is useful for scenarios where all of the relevant types are already in
+/// scope.
+///
+/// Conversely, the `foreign_path` argument is required and cannot be omitted.
+///
+/// You can also make use of `#[pallet::no_default]` on specific items in your foreign trait
+/// that you want to ensure will not be copied over but that you nonetheless want to use
+/// locally in the context of the foreign trait and the pallet (or context) in which it is
+/// defined
```

Comment:
I disagree here, it's hard to discover the `#[pallet::no_default]` attribute otherwise. You can view this as a pseudo-[helper attribute](https://doc.rust-lang.org/reference/procedural-macros.html#derive-macro-helper-attributes).